Add buy me a coffee widget and refactor some code

Not0nline opened the pull request:
What i did:
- add buy me a coffee widget
- add SRI to the script

KenKomKom commented:
Next time, if you change tests to solve it, I think it's better to mark it as green. As red, at least personally, shows that it is not yet done

Not0nline commented:
> Next time, if you change tests to solve it, I think it's better to mark it as green. As red, at least personally, shows that it is not yet done

I thought the RED means that I changed the tests and GREEN means that i changed the implementation then refactoring basically means that i dont have to change the tests but still changed the implementation
